### DISCUSSION TITLE
How to determine remote_array_1_sgs to run add_new_volume_to_storage_group?

### DISCUSSION THREAD

**Discussion by 212850a** [dell/PyU4V#136] [Q&A] [answered]
Hi,

I'm testing replication.add_new_volume_to_storage_group function for srdf_enabled storage groups and I've currently stuck - I'm not sure how I can determine remote_array_1_sgs via available RestAPI functions which is storage group name on remote array.
It is possible to get a lot of details about volumes from local (srdf_enabled) storage group - I can get rdf group number, what are their remote pair devices (on remote array), but no function I can find which would return details about storage group name of remote devices.

Do you have any idea how I can determine storage group name or remote devices from remote array with only connection to local array (without setting up new connection to remove array)?

**Comment by rawstorage** [comment, marked as answer]
Hi, there's a few ways to do this but the local api calls don't cache information about the remote devices and their storage groups.  One way is pick one of the remote volumes and look up the storage group for that volume.

presuming you have the details in a variable of one of the remote volumes, i'm using api to represent conection to PyU4V and unisphere

api.set_array_id(array_id=remote_array)
remote_sg_list = api.provisioning.get_storage_group_from_volume(volume_id='XXXXX')

**Comment by 212850a** [comment]
set_array_id(array_id=remote_array) function is exactly what I was looking for (but didn't know about it)! :)
Question is closed.
Thanks!

**Comment by rawstorage** [comment]
No worries, yeah I use this function a lot when working in multi array config, just don't forget to swtich the context back to the local array,  some functions will let you specify a different array id when running which saves you the trouble but not all.  Bit of work to do there.  Happy to help please feel free to feedback any improvements and questions through the discussions tab.  This is really where we'd like to capture conversations like this. Happy Automating and thanks for using PyU4V.
